app/src/main/java/com/example/tasky/feature_agenda/presentation/AgendaViewModel.kt

```diff
@@ -14,99 +15,96 @@ import kotlinx.coroutines.flow.MutableStateFlow
-        MutableStateFlow<AgendaViewState?>(null)
-    val viewState: StateFlow<AgendaViewState?>
-        get() = _viewState
+    private val _viewState = MutableStateFlow(AgendaViewState())
+    val viewState: StateFlow<AgendaViewState> = _viewState
 
-    // viewEvent triggered by API response
     private val _viewEvent = Channel<AgendaViewEvent>()
     val viewEvent = _viewEvent.receiveAsFlow()
 
-    // UI changes via Composable
-    private val _showDialog = MutableStateFlow(false)
-    val showDialog: StateFlow<Boolean> get() = _showDialog
+    // UI does not change
+    val initials: String = sessionStateManager.getName()?.let { ProfileUtils.getInitials(it) } ?: ""
 
-    private val _initials = MutableStateFlow<String>("")
-    val initials: StateFlow<String> get() = _initials
-
-    private val _monthSelected = MutableStateFlow<String>(getCurrentMonth())
-    val monthSelected: StateFlow<String> get() = _monthSelected
-
-    private val _dateDialogState = MutableStateFlow<MaterialDialogState>(MaterialDialogState())
-    val dateDialogState: StateFlow<MaterialDialogState> get() = _dateDialogState
+    // not displayed on screen; only necessary for business logic
+    private val _yearSelected = MutableStateFlow(LocalDate.now().year)
+    val yearSelected: StateFlow<Int> get() = _yearSelected
 
     init {
-        getProfileInitials()
+        viewModelScope.launch {
+            _viewState.emit(
+                AgendaViewState(
+                    calendarDays = calendarHelper.getCalendarDaysForMonth(
+                        LocalDate.now().year,
+                        LocalDate.now().monthValue
+                    )
+                )
+            )
+        }
     }
 
-    private val _showLogoutDropdown = MutableStateFlow(false)
-    val showLogoutDropdown: StateFlow<Boolean> get() = _showLogoutDropdown
-
     fun logOutClicked() {
         Log.d(TAG, "logOutClicked redirect user to login page")
         viewModelScope.launch {
-            _viewState.emit(AgendaViewState.LoadingSpinner)
+            _viewState.value = _viewState.value.copy(showLoadingSpinner = true)
             val result = authenticatedRemoteRepository.logOutUser()
 
             when (result) {
                 is Result.Success -> {
                     println("success logout!")
-                    // emit a viewState to show LoginScreen composable
                     _viewEvent.send(AgendaViewEvent.NavigateToLoginScreen)
                 }
 
                 is Result.Error -> {
                     println("failed logout :(")
-                    // emit a viewState to show ErrorMessage
-                    _showDialog.value = true
-                    _viewState.emit(AgendaViewState.ErrorDialog(result.error))
+                    _viewState.value = _viewState.value.copy(
+                        showLoadingSpinner = false,
+                        showErrorDialog = true,
+                        dataError = result.error
+                    )
```

Comment:
In general, I'd always use `_stateFlow.update { it.copy(...) }` to avoid race conditions. This is how it could happen:

*StateFlow holds value A*
- Thread 1 calls `copy()` on StateFlow with value A
- Before Thread 1 assigns its changed copy, Thread 2 updates StateFlow with value B
- Now, Thread 1 assigns its changed copy, which is still based on value A though, but the latest value is already value B. Therefore, value B would never be considered here.

Not always an issue, but I worked on highly stateful apps with frequently changing states, in which this really bit me, since such bugs are super hard to find.
Using `_state.value = _state.value.copy(...)` is fine if you always call this from a single thread (which you do here). But in ViewModels, you're very likely to not do that as soon as you'd change the state inside a `viewModelScope`.